### PR TITLE
Make use of mcg package and update mcg references

### PR DESCRIPTION
**Pull request by tijoytom**
Mcg now available as a package , use it in dotnet-mcg command.
Use System.CommandLine for command parsing.
Pass in AppDep and ILSDK as mcg references.

### CODE REVIEW

File: src/Microsoft.DotNet.Tools.Compiler.Native/Program.cs [hunk 1]
```diff
@@ -91,6 +82,24 @@ private static int RunMcg(NativeCompileSettings config)
...
+
+                mcgArgs.Add("--r");
+                mcgArgs.Add(refPath);
+            }
+
+            foreach (string refPath in Directory.EnumerateFiles(config.AppDepSDKPath, "*.dll"))
+            {
+                // System.Runtime.Extensions define an internal type called System.Runtime.InteropServices.Marshal which 
+                // conflicts with Marshal from S.P.Interop , we don't need System.Runtime.Extensions anyways,skip it.
+                if (refPath.Contains("System.Runtime.Extensions.dll")) continue;
```
Comment (brthor) [inline src/Microsoft.DotNet.Tools.Compiler.Native/Program.cs]: Is this a Point in Time thing or an expected permanent solution?

File: src/Microsoft.DotNet.Tools.Mcg/Program.cs [hunk 2]
```diff
@@ -40,73 +36,20 @@ private static string[] ParseResponseFile(string rspPath)
...
-                }
-            }
-            var mcgArguments = TransformCommandLine(args);
-            return ExecuteMcg(mcgArguments);
+            return ExecuteMcg(args);
         }
 
         private static int ExecuteMcg(IEnumerable<string> arguments)
         {
             // TODO : Mcg is assumed to be present under the current path.This will change once
```
Comment (brthor) [inline src/Microsoft.DotNet.Tools.Mcg/Program.cs]: Can we remove this todo?

File: src/Microsoft.DotNet.Tools.Compiler.Native/Program.cs (hunk 1 above)
Comment (tijoytom) [inline src/Microsoft.DotNet.Tools.Compiler.Native/Program.cs]: AppDep SDK is a point in time thing and is expected to go-away. Mcg do not understand packaged now , so it cant consume ILSDK package directly.But it's something which will be fixed once mcg is out in public.

File: src/Microsoft.DotNet.Tools.Mcg/Program.cs (hunk 2 above)
Comment (tijoytom) [inline src/Microsoft.DotNet.Tools.Mcg/Program.cs]: Yes ,thanks for noticing.

File: src/Microsoft.DotNet.Tools.Compiler.Native/Program.cs (hunk 1 above)
Comment (tijoytom) [inline src/Microsoft.DotNet.Tools.Compiler.Native/Program.cs]: Oh ,you are specifically asking about if ( contains check.)  ( i still haven't figured how git comments work) . yes , it's a point in time thing , we will rename the Marshal in  System.Runtime.Extensions.dll.

Comment (brthor) [inline src/Microsoft.DotNet.Tools.Compiler.Native/Program.cs]: Alright, makes sense. 

Should we put a TODO/file an issue to remove so we don't forget?

Comment (piotrpMSFT) [inline src/Microsoft.DotNet.Tools.Compiler.Native/Program.cs]: @tijoytom this is a public repo. MCG _is_ out in public ;)

Comment (piotrpMSFT) [inline src/Microsoft.DotNet.Tools.Compiler.Native/Program.cs]: @tijoytom let's please get a tracking issue before merging.

Comment (tijoytom) [inline src/Microsoft.DotNet.Tools.Compiler.Native/Program.cs]: Yes, i have the go-ahead to make mcg binaries public.
Filed  #1111

**Pull request merged**